packages/skygear-core/src/client.ts

```diff
@@ -89,7 +89,7 @@ export abstract class BaseAPIClient {
     if (this.getExtraSessionInfo) {
       const extraSessionInfo = await this.getExtraSessionInfo();
       if (extraSessionInfo) {
-        headers["x-skygear-extra-info"] = JSON.stringify(extraSessionInfo);
+        headers["x-skygear-extra-info"] = btoa(JSON.stringify(extraSessionInfo));
```

Comment:
First, nodejs does not have `btoa`

Second, it is unclear that what encoding the javascript implementation uses to encode the string into bytes. See https://stackoverflow.com/questions/30106476/using-javascripts-atob-to-decode-base64-doesnt-properly-decode-utf-8-strings

The first problem we want to solve is to first encode a javascript string into utf-8 byte sequence and then use base64 to encode the byte sequence into ascii string in browser environment.

The second problem is to inject such routine so that core is still no dependency.